src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -1032,7 +1033,12 @@ private CacheKey GenerateCacheKeyForEvict(object id, IType type, string entityOr
 					.GenerateCacheKey(id, type, entityOrRoleName);
 			}
 
-			return new CacheKey(id, type, entityOrRoleName, this);
+			if (settings.MultiTenancyStrategy != MultiTenancyStrategy.None && tenantIdentifier == null)
```

Comment:
Shouldn't it be `tenantIdentifier != null`?

Reply:
No I meant that scenario calling `ISessionFactory.EvictEntity/EvictCollection` is not implemented at all for multi-tenancy. As I moved this part to my other PR https://github.com/nhibernate/nhibernate-core/pull/2133

But now I moved logic regarding deletion object from cache per ID to this PR (and other PR is about handling whole tenant cache). Error message is adjusted accordingly